src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/MasterClusterStateUpdateStatsCollector.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    public MasterClusterStateUpdateStatsCollector(PerformanceAnalyzerController controller,
+                                                  ConfigOverridesWrapper configOverridesWrapper) {
+        super(SAMPLING_TIME_INTERVAL, MasterClusterStateUpdateStatsCollector.class.getSimpleName());
+        value = new StringBuilder();
+        this.controller = controller;
+        this.configOverridesWrapper = configOverridesWrapper;
+    }
+
+    @Override
+    public void collectMetrics(long startTime) {
+        if (!controller.isCollectorEnabled(configOverridesWrapper, getCollectorName())) {
+            return;
+        }
+        try {
+            long mCurrT = System.currentTimeMillis();
+            if (ESResources.INSTANCE.getClusterService() == null
+                    || ESResources.INSTANCE.getClusterService().getMasterService() == null) {
+                return;
+            }
+            MasterClusterStateUpdateStats currentMasterClusterStateUpdateStats = mapper.readValue(
+                    mapper.writeValueAsString(getMasterClusterStateUpdateStats()), MasterClusterStateUpdateStats.class);
+
+            MasterClusterStateUpdateMetrics masterClusterStateUpdateMetrics = new MasterClusterStateUpdateMetrics(
+                    computeLatency(currentMasterClusterStateUpdateStats), computeFailure(currentMasterClusterStateUpdateStats));
+
+            value.setLength(0);
+            value.append(PerformanceAnalyzerMetrics.getJsonCurrentMilliSeconds())
+                    .append(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor);
+            value.append(masterClusterStateUpdateMetrics.serialize());
+            saveMetricValues(value.toString(), startTime);
+
+            prevMasterClusterStateUpdateStats = currentMasterClusterStateUpdateStats;
+
+            PerformanceAnalyzerApp.WRITER_METRICS_AGGREGATOR.updateStat(
+                    WriterMetrics.MASTER_CLUSTER_UPDATE_STATS_COLLECTOR_EXECUTION_TIME, "",
+                    System.currentTimeMillis() - mCurrT);
+        } catch (Exception ex) {
+            PerformanceAnalyzerApp.ERRORS_AND_EXCEPTIONS_AGGREGATOR.updateStat(
+                    ExceptionsAndErrors.MASTER_CLUSTER_UPDATE_STATS_COLLECTOR_ERROR, "", 1);
+            LOG.debug("Exception in Collecting Master Cluster State Update Metrics: {} for startTime {}",
+                    () -> ex.toString(), () -> startTime);
+        }
+    }
+
+    @VisibleForTesting
+    public Object getMasterClusterStateUpdateStats() throws InvocationTargetException, IllegalAccessException,
+            NoSuchMethodException {
+        Method method = ClusterApplierService.class.getMethod(GET_MASTER_CLUSTER_UPDATE_STATS_METHOD_NAME);
+        return method.invoke(ESResources.INSTANCE.getClusterService().getMasterService());
+    }
+
+    /**
+     * MasterClusterUpdateMetric is ES is a tracker for total time taken to publish cluster state and the
```

Comment:
nit: MasterClusterUpdateMetric **in** ES ...

Reply:
done